Small cleanups and one performance improvement

ptroja opened the pull request.

ikazuhiro commented:
1. Please keep let-binding of `pgg-decrypt-automatically`.  Wanderlust still supports SEMI (not SEMI-EPG) with PGG.
2. `wl-create-access-folder-entity` does not seem equivalent for me.  The original modifies only `car` of `flists`.
3. In `elmo-get-file-string`, it seems better that `(when (file-exists-p filename)` line moved to out of `elmo-set-work-buf`.
4. In `wl-thread-insert-top`, the second `(length wl-thread-entity-list)` should be replaced with `len`.

ptroja commented:
> Please keep let-binding of pgg-decrypt-automatically. Wanderlust still supports SEMI (not SEMI-EPG) with PGG.

Done.

> wl-create-access-folder-entity does not seem equivalent for me. The original modifies only car of flists.

I need to investigate. For now I just dropped this commit from my master branch (and so from the pull request too).

> In elmo-get-file-string, it seems better that (when (file-exists-p filename) line moved to out of elmo-set-work-buf.

Done. I added an extra IF to return an emtpy string if the file does not exist (otherwise this routine would return nil and string-match on the caller site would raise an error).

> In wl-thread-insert-top, the second (length wl-thread-entity-list) should be replaced with len.

Done.

ikazuhiro commented:
Merged, thank you for your contribution.
